clientv3/compare.go

```diff
@@ -99,6 +99,18 @@ func (cmp *Cmp) ValueBytes() []byte {
 // WithValueBytes sets the byte slice for the comparison's value.
 func (cmp *Cmp) WithValueBytes(v []byte) { cmp.TargetUnion.(*pb.Compare_Value).Value = v }
 
+// WithRange sets the comparison to scan the range [key, end).
+func (cmp Cmp) WithRange(end string) Cmp {
```

Comment:
I tried it that way but `clientv3.Compare(...).WithRange(...)` will complain that the result from `Compare()` is not a pointer and it doesn't seem worth changing Compare()'s return value to a pointer to support this